cmdstanpy/utils.py

```diff
@@ -206,6 +210,7 @@ def scan_column_names(fp: TextIO, config_dict: Dict, lineno: int) -> int:
     lineno += 1
     names = line.split(',')
     config_dict['column_names'] = tuple(names)
+    config_dict['num_params'] = len(names) - 1
```

Comment:
Is that -1 for `lp__`?

Reply:
@mitzimorris / @ahartikainen this is only the default value for `'num_params'`. In sampling it gets replaced with original implementation, don't you think it is enough to subtract `lp__` from total number of columns to get param length? 
Another question is how should we count vector params, as a single length or total (see my implementation of `optimized_params_dict` - it mimics pystan behavior)?

Reply:
OK w/ implementation for 'num_params'

container type parameters - vector, matrix, and arrays - are counted by the number of elements - this will match the dimension of the mass matrix.

I'm not sure about `optimized_params_dict` -

the csv output reflects the names in the c++ program - e.g. if you have parameters `matrix[2][2] foo` then the generated c++ model names are `foo.1.1` ... `foo.2.2`

I think it would be good to be able to access `foo.1.2` as well as `foo`.  looks like the way that `optimized_params_dict` is set up you lose the column names?  and therefore all container type parameters get flattened into a vector?

cf implementation `get_drawset` - https://github.com/stan-dev/cmdstanpy/blob/dd9e7aba44cc2f64861aaa715ced00ba3b05f7a6/cmdstanpy/stanfit.py#L393-L397 

it's a bit of a grey area here.  my direct experience with vector parameters was working with the model whose output is used as test data - https://github.com/stan-dev/cmdstanpy/tree/60-implement-optimize/test/data/runset-big  - when working with that model in RStan, I usually looked at just the first and last few elements of the parameter.

Reply:
Reverted to previous implementation.